cves/kernel/CVE-2021-3640.yml

```diff
@@ -381,8 +397,10 @@ order_of_operations:
     Answer must be true or false.
     Write a note about how you came to the conclusions you did, regardless of
     what your answer was.
-  answer:
-  note:
+  answer: true
+  note: |
+    The fix involves correcting an order of operations because there is a check 
+    before the original code to prevent the vulnerability. 
 lessons:
```

Comment:
I think defense in depth would apply here. By adding extra checks to make sure that resources were getting properly malloced, freed, and synchronized this issue could have been prevented.